runtime/sema/type.go

```diff
@@ -4624,6 +4642,82 @@ func (t *PublicAccountType) Resolve(_ map[*TypeParameter]Type) Type {
 	return t
 }
 
+// AccountKeyType
+
+type AccountKeyType struct{}
+
+func (*AccountKeyType) IsType() {}
+
+func (*AccountKeyType) String() string {
+	return "AccountKey"
+}
+
+func (*AccountKeyType) QualifiedString() string {
+	return "AccountKey"
+}
+
+func (*AccountKeyType) ID() TypeID {
+	return "AccountKey"
+}
+
+func (*AccountKeyType) Equal(other Type) bool {
+	_, ok := other.(*AccountKeyType)
+	return ok
+}
+
+func (*AccountKeyType) IsResourceType() bool {
+	return false
+}
+
+func (*AccountKeyType) TypeAnnotationState() TypeAnnotationState {
+	return TypeAnnotationStateValid
+}
+
+func (*AccountKeyType) IsInvalidType() bool {
+	return false
+}
+
+func (*AccountKeyType) ContainsFirstLevelInterfaceType() bool {
+	return false
+}
+
+func (*AccountKeyType) CanHaveMembers() bool {
+	return true
+}
+
+func (t *AccountKeyType) GetMember(identifier string, _ ast.Range, _ func(error)) *Member {
+	newField := func(fieldType Type) *Member {
+		return NewPublicConstantFieldMember(t, identifier, fieldType)
+	}
+
+	switch identifier {
+	case "publicKey":
+		return newField(&VariableSizedType{Type: &Int8Type{}})
+
+	case "signingAlgorithm":
+		return newField(&Int8Type{})
+
+	case "hashingAlgorithm":
+		return newField(&Int8Type{})
+
+	case "weight":
+		return newField(&UInt64Type{})
+
+	default:
+		return nil
+	}
+}
+
+func (t *AccountKeyType) Unify(_ Type, _ map[*TypeParameter]Type, _ func(err error), _ ast.Range) bool {
+	// return false
+	return false
+}
+
+func (t *AccountKeyType) Resolve(_ map[*TypeParameter]Type) Type {
+	// return t
+	return nil
```

Comment:
Should this be `return t`?

Reply:
Yep, should return itself, nothing to resolve